### PR TITLE
Allow only READONLY_USERNAME and READONLY_PASSWORD to be set for auth

### PR DESCRIPTION
**Pull request by jingyuanliang**
This is useful for registries that are supposed to be just mirrors of upstream registries (using REGISTRIES_JSON).

### CODE REVIEW

File: src/authentication-method.ts [hunk 1]
```diff
@@ -6,19 +6,20 @@ import type { AuthenticatorCredentials } from "./user";
 export async function authenticationMethodFromEnv(env: Env) {
   if (env.JWT_REGISTRY_TOKENS_PUBLIC_KEY) {
     return await newRegistryTokens(env.JWT_REGISTRY_TOKENS_PUBLIC_KEY);
```
Comment (gabivlj) [inline src/authentication-method.ts:8]: nit: can we make this be
`(env.USERNAME && env.PASSWORD) || (env.READONLY_USERNAME && env.READONLY_PASSWORD)`

IK the logic is correct but just trying to make it a bit more obvious that it's doing the right thing first hand.

**Review by gabivlj — CHANGES_REQUESTED** [review]
Looks great, thanks for the contribution. Just one minor suggestion!

File: src/authentication-method.ts [hunk 2]
```diff
@@ -6,19 +6,20 @@ import type { AuthenticatorCredentials } from "./user";
...
-  } else if (env.USERNAME && env.PASSWORD) {
-    const credentials: AuthenticatorCredentials[] = [
-      { username: env.USERNAME, password: env.PASSWORD, capabilities: ["pull", "push"] }
-    ];
+  } else if (env.USERNAME && env.PASSWORD || env.READONLY_USERNAME && env.READONLY_PASSWORD) {
```
Comment (gabivlj) [inline src/authentication-method.ts:9]: nit: can we make this be
(env.USERNAME && env.PASSWORD) || (env.READONLY_USERNAME && env.READONLY_PASSWORD)

IK the logic is correct but just trying to make it a bit more obvious that it's doing the right thing first hand.

**Comment by jingyuanliang** [comment]
uhm I found the new test case not effective. I mean, it still passes without the src/authentication-method.ts change. Any idea?

**Comment by gabivlj** [comment]
Oh yeah. Good catch.
https://github.com/jingyuanliang/serverless-registry/blob/ea743e310b5c4a7f87b5da6956567b5f1590434d/test/index.test.ts#L117-L135

Probably take inspiration of this test. I think the manifest test doesn't take the auth path as it uses the client directly!

**Comment by jingyuanliang** [comment]
Regardless of auth, I wanted to test that it starts without error at least...

**Pull request merged**